core/src/main/java/org/testcontainers/containers/FixedHostPortGenericContainer.java

```diff
@@ -17,14 +18,26 @@ public FixedHostPortGenericContainer(@NotNull String dockerImageName) {
     }
 
     /**
-     * Bind a fixed port on the docker host to a container port
+     * Bind a fixed TCP port on the docker host to a container port
      * @param hostPort          a port on the docker host, which must be available
      * @param containerPort     a port in the container
      * @return                  this container
      */
     public SELF withFixedExposedPort(int hostPort, int containerPort) {
 
-        super.addFixedExposedPort(hostPort, containerPort);
+        return withFixedExposedPort(hostPort, containerPort, InternetProtocol.DEFAULT);
+    }
+
+    /**
+     * Bind a fixed port on the docker host to a container port
+     * @param hostPort          a port on the docker host, which must be available
+     * @param containerPort     a port in the container
+     * @param protocol          an internet protocol (tcp or udp)
+     * @return                  this container
+     */
+    public SELF withFixedExposedPort(int hostPort, int containerPort, InternetProtocol protocol) {
```

Comment:
I'd rather not leak docker-java implementation details into the public API. Instead I think we should use our own Enums here.

Reply:
Good point @kiview 
Fixed.